ovs/extensions/db/arakoon/tests/test_arakooninstaller.py

```diff
@@ -0,0 +1,226 @@
+#      http://www.openvstorage.com.
+#
+# This file is free software; you can redistribute it and/or modify it
+# under the terms of the GNU Affero General Public License v3 (GNU AGPLv3)
+# as published by the Free Software Foundation, in version 3 as it comes
+# in the LICENSE.txt file of the Open vStorage OSE distribution.
+#
+# Open vStorage is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY of any kind.
+
+"""
+Module for testing the ArakoonInstaller
+"""
+import os
+import json
+import shutil
+import unittest
+from ovs.dal.hybrids.servicetype import ServiceType
+from ovs.extensions.generic.configuration import Configuration
+from ovs.extensions.generic.system import System
+from ovs.extensions.generic.sshclient import SSHClient
+from ovs.extensions.storage.persistentfactory import PersistentFactory
+from ovs.extensions.storage.volatilefactory import VolatileFactory
+from ovs.extensions.db.arakoon.ArakoonInstaller import ArakoonInstaller, ArakoonClusterConfig
+from ovs.lib.tests.helpers import Helper
+
+
+class ArakoonInstallerTester(unittest.TestCase):
+    """
+    This test class will validate the various scenarios of the MDSService logic
+    """
+    @classmethod
+    def setUpClass(cls):
+        """
+        Sets up the unittest, mocking a certain set of 3rd party libraries and extensions.
+        This makes sure the unittests can be executed without those libraries installed
+        """
+        cls.persistent = PersistentFactory.get_client()
+        cls.persistent.clean()
+
+        cls.volatile = VolatileFactory.get_client()
+        cls.volatile.clean()
+
+    @classmethod
+    def tearDownClass(cls):
+        """
+        Tear down changes made during setUpClass
+        """
+        Configuration._unittest_data = {}
+
+        cls.persistent = PersistentFactory.get_client()
+        cls.persistent.clean()
+
+        cls.volatile = VolatileFactory.get_client()
+        cls.volatile.clean()
+
+    def setUp(self):
+        """
+        (Re)Sets the stores on every test
+        """
+        self.persistent.clean()
+        self.volatile.clean()
+        self.maxDiff = None
+
+    def tearDown(self):
+        """
+        Clean up test suite
+        """
+        self.persistent.clean()
+        self.volatile.clean()
+
+    def test_cluster_maintenance(self):
+        """
+        Validates whether a cluster can be correctly created
+        """
+        Configuration.set('/ovs/framework/hosts/1/ports', {'arakoon': [10000, 10100]})
+        Configuration.set('/ovs/framework/hosts/2/ports', {'arakoon': [20000, 20100]})
+
+        structure = Helper.build_service_structure(
+            {'storagerouters': [1, 2]}
+        )
+        storagerouters = structure['storagerouters']
+        System._machine_id = {storagerouters[1].ip: '1',
+                              storagerouters[2].ip: '2'}
+
+        # Create new cluster
+        mountpoint = storagerouters[1].disks[0].partitions[0].mountpoint
```

Comment:
If we ever make changes to the mountpoint of the default partition layout, we could potentially end up doing ```shutil.rmtree('/')```